### PR TITLE
Introduce Redis into CI and add memoizer suite

### PR DESCRIPTION
**Pull request by khcoaching**
Sets up a Redis instance for CI and introduces a basic suite for the memoizer. Tests run against the last redis db and prod / dev run against the first one. That way we can call `flushdb` without blowing away local cache if you're running a local redis. 

Oh and you must run a local redis to run the server specs now. We can setup a container that starts with the tests at some point.

### CODE REVIEW

**Comment by khcoaching** [comment]
@kkennis @luckyOdisetti 

@zmanian for viz

File: packages/server/src/util/memoizer-redis/lock.ts [hunk 1]
```diff
@@ -15,12 +15,14 @@ export default function lockFactory(client: Redis.Redis, options: LockOptions):
...
   // locking fn
-  return async function lock(lockKey: string, timeout = lockTimeout, retry = lockRetry): Promise<Unlocker> {
+  return async function lock(lockKey: string, timeout = lockTimeout, retry = lockRetry): Promise<Unlocker | void> {
     lockKey = `lock:${lockKey}`;
     const expireAt = Date.now() + timeout + 1;
 
     // try to set a lock
-    await setLock(client, lockKey, expireAt, retry);
+    const aquired = await setLock(client, lockKey, expireAt, retry);
+    if (!aquired) return;
```
Comment (khcoaching) [inline packages/server/src/util/memoizer-redis/lock.ts]:
```suggestion
    if (!acquired) return;
```

File: packages/server/src/util/memoizer-redis/lock.ts [hunk 2]
```diff
@@ -15,12 +15,14 @@ export default function lockFactory(client: Redis.Redis, options: LockOptions):
...
 
   // locking fn
-  return async function lock(lockKey: string, timeout = lockTimeout, retry = lockRetry): Promise<Unlocker> {
+  return async function lock(lockKey: string, timeout = lockTimeout, retry = lockRetry): Promise<Unlocker | void> {
     lockKey = `lock:${lockKey}`;
     const expireAt = Date.now() + timeout + 1;
 
     // try to set a lock
-    await setLock(client, lockKey, expireAt, retry);
+    const aquired = await setLock(client, lockKey, expireAt, retry);
```
Comment (khcoaching) [inline packages/server/src/util/memoizer-redis/lock.ts]:
```suggestion
    const acquired = await setLock(client, lockKey, expireAt, retry);
```

File: packages/server/test/util/memoizer-redis/index.spec.ts [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+import memoizer, {
+  defaultOptions,
+  sha1,
+  getFnNamespace,
+  getCacheKey,
+  lookup,
+  serialize,
+  deserialize,
+} from 'util/memoizer-redis/index';
+import { escapeHTML } from 'core-js/core/string';
```
Comment (khcoaching) [inline packages/server/test/util/memoizer-redis/index.spec.ts]:
```suggestion
```

**Comment by kkennis** [comment]
Don't see anything wrong, let's merge

**Pull request merged**